serpentTools/parsers/branching.py

```diff
@@ -115,3 +119,17 @@ def iterBranches(self):
         """Iterate over branches yielding paired branch IDs and containers"""
         for bID, b in iteritems(self.branches):
             yield bID, b
+
+    def _precheck(self):
+        """todo: is there a good precheck to do here?
+        """
+        pass
```

Comment:
Possibly. We can determine, [from the first few lines](http://serpent.vtt.fi/mediawiki/index.php/Automated_burnup_sequence#Example_output_block), how many branches and universes *should* be present in the output file. You do this above with the `sefl._totalBranches` already. If we put that information, as well as the total coefficients stored, in the `_precheck` method, we don't have to check `if self._totalBranches == None` in the main `_read` method.

Reply:
Alright, I like that idea for using precheck. So, I added that. What do you mean by total coefficients stored though? I think if Serpent quits early, you're almost guaranteed to have complete branch output, but missing entire branches. This is because that output part of Serpent is probably 1/100000 of the runtime.

Reply:
Total coefficients I guess would be the sum of all coefficients requested times the number of universes. What you have works well, and I added comments in the review